### PR TITLE
Improve URI comparison logic

### PR DESCRIPTION
**Pull request by mykhailopopivniak**
- Extract repetetive code into separate functions to improve maintainability and extensibility.
- According to RFC 3261 and RFC5627, `gr` parameter should be considered in URI comparison if it appears in both compared URIs.
-  Consider URI parameters in Uri's `operator<` to ensure correct logic, consistent with Uri's `operator==`.

### CODE REVIEW

**Comment by mykhailopopivniak** [comment]
Hi, @sgodin. Could you review this PR when you have time? It would be great to get this included in the next release. Let me know if there's anything that needs adjustment. Thanks!

**Comment by sgodin** [comment]
@mykhailopopivniak Will do!  I want to make sure all TFM tests pass first, then I'll merge in and make sure no tests are broken.

File: resip/stack/Uri.cxx [hunk 1]
```diff
@@ -748,7 +703,95 @@ Uri::operator<(const Uri& other) const
...
+      unB.insert(*i);
+   }
+#else
+   ParameterList unA = mUnknownParameters;
+   ParameterList unB = other.mUnknownParameters;
+
+   sort(unA.begin(), unA.end(), orderUnknown);
+   sort(unB.begin(), unB.end(), orderUnknown);
+#endif
+
```
Comment (sgodin) [inline resip/stack/Uri.cxx:759]: Can you please add some comments in the code explaining what this section of code is doing?  ie:  Why build up 2 new sets?  Why is code different for different platforms?  And a general explanation of the logic below.  :)  Thank you in advance.

Comment (mykhailopopivniak) [reply resip/stack/Uri.cxx:765]: Added comments and renamed some variables to make it easier to understand.
This part of the code mirrors the one from `Uri::operator==`.

**Pull request merged**